app/components/file_attachment_list/file_attachment_list.js

```diff
@@ -124,61 +121,130 @@ export default class FileAttachmentList extends Component {
-                    file={f}
-                    id={file.id}
-                    index={idx}
-                    onCaptureRef={this.handleCaptureRef}
-                    onPreviewPress={this.handlePreviewPress}
-                    onLongPress={this.props.onLongPress}
-                    theme={this.props.theme}
-                />
+                >
+                    <FileAttachment
+                        key={file.id}
+                        canDownloadFiles={canDownloadFiles}
+                        file={f}
+                        id={file.id}
+                        index={this.attachmentIndex(file.id)}
+                        onCaptureRef={this.handleCaptureRef}
+                        onPreviewPress={this.handlePreviewPress}
+                        onLongPress={this.props.onLongPress}
+                        theme={this.props.theme}
+                        imagesListProps={imagesListProps}
+                    />
+                </View>
             );
         });
     };
 
+    renderImageRow = (images) => {
+        if (images.length === 0) {
+            return null;
+        }
+
+        const imagesSet = (images.length > MAX_VISIBLE_ROW_IMAGES) ? images.slice(0, MAX_VISIBLE_ROW_IMAGES) : images;
+
+        return (
+            <View>
+                <View style={styles.row}>
+                    { this.renderItems(imagesSet) }
+                </View>
+            </View>
+        );
+    };
+
     render() {
-        const {fileIds, isFailed} = this.props;
+        const {canDownloadFiles, fileIds, files, isFailed} = this.props;
 
+        if (!files.length && fileIds.length > 0) {
+            return fileIds.map((id, idx) => (
+                <FileAttachment
+                    key={id}
+                    canDownloadFiles={canDownloadFiles}
+                    file={{loading: true}}
+                    id={id}
+                    index={idx}
+                    theme={this.props.theme}
+                />
+            ));
+        }
+
+        const images = this.imagesSet(files);
+        const nonImages = this.nonImagesSet(files);
         return (
-            <ScrollView
-                horizontal={true}
-                scrollEnabled={fileIds.length > 1}
-                style={[(isFailed && styles.failed)]}
-                keyboardShouldPersistTaps={'always'}
-            >
-                {this.renderItems()}
-            </ScrollView>
+            <View style={isFailed && styles.failed}>
+                {this.renderImageRow(images)}
+                {this.renderItems(nonImages)}
+            </View>
         );
     }
 }
 
 const styles = StyleSheet.create({
+    row: {
+        flex: 1,
+        flexDirection: 'row',
+    },
+    image: {
+        position: 'absolute',
+        width: '100%',
+        height: '100%',
+        resizeMode: 'cover',
+        top: 0,
+        left: 0,
+    },
+    pictureFrame: {
+        width: '100%',
+        flexShrink: 1,
+        margin: 3,
```

Comment:
@enahum This one has had me running around for alternatives, but I'm coming up stumped.

Since we have to use `StyleSheet.absoluteFill` along with the `paddingBottom: 100%` setting, the image thumbnails in the attachments row squeeze together to ensure they all fit within the allowed flex space, from where the width and height get measured for eventual rendering.

Changing the lines above to:

```
row: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
},

pictureFrame: {
    flex: 1,
    // margin: 3,
},
```

and exaggerating/red highlighting the borderRadius of the rendered images for effect, we still see the following result:

<img width="356" alt="Screen Shot 2019-11-06 at 23 05 30" src="https://user-images.githubusercontent.com/887849/68354195-7431f600-00ea-11ea-8079-020b9e4bb8a5.png">

In other words, because of the automated squeezing, flex has no extra space to offer to 'space-between'.

Could I recommend we continue with a margin setting (perhaps just 2 points, instead of 3?) for the time-being, and reconcile the small initial margin for image attachment rows as a future item?

Reply:
> What about applying a left margin to all the images except the first?

You bet! Implemented via 230d2a1169cb3496591f26f0337151ed0e1f9c7c .